pkg/host/configvalidation.go

```diff
@@ -0,0 +1,322 @@
+	return true, true, nil
+}
+
+// AdvancedPCISettingsEnabled returns true if ADVANCED_PCI_SETTINGS param is enabled for current config
+func (v *configValidationImpl) AdvancedPCISettingsEnabled(currentConfig map[string]string) bool {
+	if value, found := currentConfig[consts.AdvancedPCISettingsParam]; found && value == consts.NvParamTrue {
+		return true
+	}
+	return false
+}
+
+// RuntimeConfigApplied checks if desired runtime config is applied
+func (v *configValidationImpl) RuntimeConfigApplied(device *v1alpha1.NicDevice) (bool, error) {
+	ports := device.Status.Ports
+
+	desiredMaxReadReqSize, desiredTrust, desiredPfc := v.CalculateDesiredRuntimeConfig(device)
+
+	if desiredMaxReadReqSize != 0 {
+		actualMaxReadReqSize, err := v.utils.GetMaxReadRequestSize(ports[0].PCI)
+		if err != nil {
+			log.Log.Error(err, "can't validate maxReadReqSize", "device", device.Name)
+			return false, err
+		}
+		if actualMaxReadReqSize != desiredMaxReadReqSize {
+			return false, nil
+		}
+
+		if len(ports) == 2 {
+			actualMaxReadReqSize, err = v.utils.GetMaxReadRequestSize(ports[1].PCI)
+			if err != nil {
+				log.Log.Error(err, "can't validate maxReadReqSize", "device", device.Name)
+				return false, err
+			}
+			if actualMaxReadReqSize != desiredMaxReadReqSize {
+				return false, nil
+			}
+		}
+	}
+
+	actualTrust, actualPfc, err := v.utils.GetTrustAndPFC(ports[0].NetworkInterface)
+	if err != nil {
+		log.Log.Error(err, "can't validate QoS settings", "device", device.Name)
+		return false, err
+	}
+	if actualTrust != desiredTrust || actualPfc != desiredPfc {
+		return false, nil
+	}
+	if len(ports) == 2 {
+		actualTrust, actualPfc, err := v.utils.GetTrustAndPFC(ports[1].NetworkInterface)
+		if err != nil {
+			log.Log.Error(err, "can't validate QoS settings", "device", device.Name)
+			return false, err
+		}
+		if actualTrust != desiredTrust || actualPfc != desiredPfc {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+// CalculateDesiredRuntimeConfig returns desired values for runtime config
+// returns int - maxReadRequestSize
+// returns string - qos trust mode
+// returns string - qos pfc settings
+// returns err - an error occurred
+func (v *configValidationImpl) CalculateDesiredRuntimeConfig(device *v1alpha1.NicDevice) (int, string, string) {
+	maxReadRequestSize := 0
+	trust := ""
+	pfc := ""
+
+	template := device.Spec.Configuration.Template
+
+	if template.PciPerformanceOptimized != nil && template.PciPerformanceOptimized.Enabled {
+		if template.PciPerformanceOptimized.MaxReadRequest != 0 {
+			maxReadRequestSize = template.PciPerformanceOptimized.MaxReadRequest
+		} else {
+			maxReadRequestSize = 4096
+		}
+	}
+
+	if template.RoceOptimized != nil && template.RoceOptimized.Enabled {
+		if template.RoceOptimized.Qos != nil {
+			trust = template.RoceOptimized.Qos.Trust
+			pfc = template.RoceOptimized.Qos.PFC
+		} else {
+			trust = "dscp"
+			pfc = "0,0,0,1,0,0,0,0"
+		}
+	} else {
+		trust = "pcp"
+		pfc = "0,0,0,0,0,0,0,0"
+	}
```

Comment:
nit: You could set this as the default values while initializing the vars and remove the else block.